instruments/abstract_instruments/comm/usb_communicator.py

```diff
@@ -77,25 +111,47 @@ def close(self):
         """
         Shutdown and close the USB connection
         """
-        try:
-            self._conn.shutdown()
-        finally:
-            self._conn.close()
+        self._dev.reset()
+        usb.util.dispose_resources(self._dev)
 
-    def read_raw(self, size=-1):
-        raise NotImplementedError
+    def read_raw(self, size=1000):
+        """Read raw string back from device and return.
```

Comment:
Choosing `size=1000` is a weird way of doing this, but I don't have a good idea right now on how else to specify it. Need to think more and try some ideas. Let me know if you have any :)

Reply:
See below